aten/src/ATen/native/cuda/LogcumsumexpKernel.cu

```diff
@@ -11,24 +11,75 @@
 
 namespace at::native {
 
+// custom min and max to be used in logcumsumexp for complex arguments
+template <typename scalar_t>
+__host__ __device__ c10::complex<scalar_t> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y, bool min) {
+  scalar_t xr = std::real(x);
```

Comment:
the CPU and CUDA code seem to be identical. It is possible to solve that. You could have a look at how we accomplish that in sparse. You could have a look at `git grep FUNCAPI`.

Reply:
Thanks @nikitaved. I'd like to get this PR merged first and I can do this later.